timetracker/api/views.py

```diff
@@ -1,22 +1,187 @@
+import json
+from datetime import timedelta
+from datetime import datetime
+
+from django.db.models import Sum
+from rest_framework import status
+from rest_framework.decorators import action, permission_classes
 from rest_framework.viewsets import ModelViewSet
-from timetracker.models import Employee
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.decorators import api_view
+from timetracker.models import Employee, WorkingHours
 from timetracker.models import Checking
-from timetracker.models import Vacation
-from .serializers import EmployeeSerializer
+from .serializers import EmployeeSerializer, WorkingHoursSerializer
 from .serializers import CheckingSerializer
 from .serializers import VacationSerializer
 
 
 class EmployeeViewSet(ModelViewSet):
+    permission_classes = [IsAuthenticated]
     queryset = Employee.objects.all()
     serializer_class = EmployeeSerializer
 
+    YEAR_QUARTERS = {
+        '1': [1, 3],
+        '2': [3, 6],
+        '3': [6, 9],
+        '4': [9, 12]
+    }
+
+    @action(methods=['POST'], detail=True, permission_classes=[IsAuthenticated],
+            url_path='set-check', url_name='set_check')
+    def set_checking(self, request, pk=None):
+        check = request.data['check'] if 'check' in request.data else ''
```

Comment:
This can be replaced by:

```suggestion
        check = request.data.get('check', '')
```